web-services/accumulo/src/main/java/datawave/webservice/operations/remote/RemoteStatsService.java

```diff
@@ -0,0 +1,41 @@
+package datawave.webservice.operations.remote;
+
+import com.codahale.metrics.annotation.Timed;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector;
+import com.fasterxml.jackson.module.jaxb.JaxbAnnotationModule;
+import datawave.configuration.RefreshableScope;
+import datawave.webservice.response.StatsResponse;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.Priority;
+import javax.enterprise.inject.Alternative;
+import javax.interceptor.Interceptor;
+
+@RefreshableScope
+@Alternative
+@Priority(Interceptor.Priority.APPLICATION)
+public class RemoteStatsService extends RemoteAccumuloService {
+    
+    private static final String STATS_SUFFIX = "stats";
+    
+    private ObjectMapper statsMapper;
+    private ObjectReader statsReader;
+    
+    @Override
+    @PostConstruct
+    public void init() {
+        super.init();
+        statsMapper = new ObjectMapper();
+        statsMapper.registerModule(new JaxbAnnotationModule());
+        statsMapper.setAnnotationIntrospector(new JaxbAnnotationIntrospector(TypeFactory.defaultInstance()));
+        statsReader = statsMapper.readerFor(StatsResponse.class);
```

Comment:
RemoteHttpService, which this extends via RemoteAccumuloService includes an objectMapper member that is configured to handle JAX-B annotations, Guava objects (e.g., Multimap), and also a property that configures the jackson mapper to use XmlElementWrapper names as the property name. You need the last one for complete compatibility with previous responses. Is there a reason you aren't using this objectMapper member here and in your other RemoteXXXService classes?

Reply:
In the first iteration, I did make use of that mapper, but when the 2.8.x vs. 2.7.x Jackson incompatibilities with Wildfly 10 surfaced in my testing, my main concern was isolating RemoteHttpService from all my trial/error, particularly since I needed additional mapper configuration, e.g., to map the abstract KeyBase to DefaultKey for LookupResponse. I'll revisit.

Although, it occurs to me now that I could simplify the code quite a bit, and also protect against similar Jackson versioning issues in the future, by just requesting application/xml explicitly from the microservice and then just using simple unmarshal via built-in JAXBContext, no Jackson required. That might require additional hoop to jump thru, which is that the XML responses from the microservice won't be XMLNS-qualified, as would be required for deserialization on the Wildfly side. But that's easily solved with a custom XMLFilter to manage the SAX parse, to inject the expected namespace, which would only be a few additional lines of code. The existing StatsBean code does something similar for the Accumulo monitor's response, which is why I thought of it

Thoughts?

Reply:
Updated in latest commit to use RemoteHttpService's objectmapper, which is now configured via CDI bean